modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -323,4 +326,16 @@ private void validateConfigurationProviderIsNotExpression() throws Initialisatio
   protected ConfigurationProvider getConfigurationProvider() {
     return configurationProvider;
   }
+
+  private ParameterValueResolver getMetadataKeyValueResolver(MetadataKey key) {
```

Comment:
this case should be handled by the mediator on its own. It has all the required information to resolve the value of the key. We can add a method that receives (context, key) in the mediator, and that method generates this resolver to delegate in the method expecting a PArameterValueResolver